### PR TITLE
PLG-40: Revamp the Product Gallery row by using the Card component from the DSM

### PR DESCRIPTION
**Pull request by olivierpontier**
**Description (for Contributor and Core Developer)**

This pull requests aims to homogenize the render of the Product Gallery Grid elements by using DSM components.

preview

- [x] RAC-527 dependency: https://github.com/akeneo/pim-community-dev/pull/13984/

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/ProductGalleryRow.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      const route = this.isProductModel() ? 'pim_enrich_product_model_edit' : 'pim_enrich_product_edit';
+      Router.redirectToRoute(route, {
+        id: technicalId
+      });
+    };
+
+    const selectProduct = (isSelected: boolean) => {
+      this.model.trigger('backgrid:selected', this.model, isSelected);
+
+      // @fixme: the rerender cause the following warning in the console  "Warning: render(...): It looks like the React-rendered content of this container was removed without using React. This is not supported and will cause errors. Instead, call ReactDOM.unmountComponentAtNode to empty a container"
```
Comment (juliensnz) [inline src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/ProductGalleryRow.tsx:51]: Will you fix this before merge?

File: src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/ProductGalleryRow.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    return MediaUrlGenerator.getMediaShowUrl(image.filePath, 'thumbnail');
+  }
+
+  private getBadgeText() {
+    if (this.isProductModel()) {
+      const complete = this.model.get('complete_variant_products').complete;
+      const total = this.model.get('complete_variant_products').total;
+
+      return complete + ' / ' + total;
```
Comment (juliensnz) [inline src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/ProductGalleryRow.tsx:100]:
```suggestion
      return `${complete}/${total}`;
```

File: src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/ProductGalleryRow.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    const completeness = this.model.get('completeness');
+    if (null !== completeness) {
+      return completeness + '%';
```
Comment (juliensnz) [inline src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/ProductGalleryRow.tsx:105]:
```suggestion
      return `${completeness}%`;
```

**Review by juliensnz — APPROVED** [review]

File: src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/ProductGalleryRow.tsx (hunk 1 above)
Comment (olivierpontier) [reply src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/ProductGalleryRow.tsx]: Yes, I will. I didn't find the source of the problem for now. The call to the `React.unmountComponentAtNode` does not work and crashes the page on loading. A lead could be the parents Grid and Grid.body elements update the DOM before the rerender of the Row. What do you think about it? Do you have already face that kind of issue?

**Review by AymericPlanche — APPROVED** [review]

**Review by LaurentPetard — APPROVED** [review]

**Pull request merged**